src/lib.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2017-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#![warn(missing_docs)]
+
+//! List of well-known SS58 account types as an enum.
+use core::convert::TryFrom;
+
+include!(concat!(env!("OUT_DIR"), "/account_type_enum.rs"));
+
+/// Error encountered while parsing `Ss58AddressFormat` from &'_ str
+/// unit struct for now.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub struct ParseError;
+
+/// A custom address format. See also KnownSs58AddressFormat
+#[non_exhaustive]
```

Comment:
```suggestion
```

Makes no sense here.

Reply:
Interesting, I was thinking this was exactly the situation that one would use the attribute in.
When we add an additional known address types it would be backwardly compatible with the attribute there. Otherwise each additional type is technically a breaking change. I suspect most matches would have an arm matching against `_` anyway, but this just ensures that arm is there.

Reply:
This here is the struct, not the enum. For the enum you are right, for the struct it is not required.

Reply:
Ah sorry - sometimes github doesn't give much context. I assumed it was on the enum - good catch!